File: frontend/src/pages/documentation/mcp-reference/data/documentation.ts
// Copyright (c) 2026 Valaris Studio
// SPDX-License-Identifier: AGPL-3.0-or-later

import type { ToolDoc } from "./types";

export const DOCUMENTATION_TOOL_DOCS: ToolDoc[] = [
  {
    name: "list_documentation",
    category: "context",
    kind: "read",
    description: "List product documentation from the connected platform, including version and translation status.",
    params: [
      {
        name: "locale",
        required: false,
        description: "Registered documentation locale: en, es or pt-BR."
      },
      {
        name: "offset",
        required: false,
        description: "Section offset, starting at zero."
      },
      {
        name: "limit",
        required: false,
        description: "Maximum sections to return (1–100)."
      }
    ],
    gotchas: [
      "Product documentation contains no workspace data. Normal authentication and MCP allowlists still apply."
    ],
    examplePrompt: "List the documentation available on this platform in <locale>.",
    related: [
      "read_documentation"
    ]
  },
  {
    name: "read_documentation",
    category: "context",
    kind: "read",
    description: "Read a bounded Markdown section from the connected platform.",
    params: [
      {
        name: "slug",
        required: true,
        description: "Section slug returned by list_documentation."
      },
      {
        name: "locale",
        required: false,
        description: "Registered documentation locale: en, es or pt-BR."
      },
      {
        name: "version",
        required: false,
        description: "Version returned by list_documentation; a mismatch fails explicitly."
      },
      {
        name: "offset",
        required: false,
        description: "Character offset for continuing a section."
      },
      {
        name: "limit",
        required: false,
        description: "Maximum characters to return (1–30000)."
      }
    ],
    gotchas: [
      "Follow next_offset with the same locale and version. No bundled documentation fallback is used.",
      "Product documentation contains no workspace data. Normal authentication and MCP allowlists still apply."
    ],
    examplePrompt: "Read <slug> using the version from list_documentation, continuing until next_offset is null.",
    related: [
      "list_documentation"
    ]
  }
];

File: mcp-server/src/valaris_mcp/tools/documentation.py
# Copyright (c) 2026 Valaris Studio
# SPDX-License-Identifier: AGPL-3.0-or-later

from __future__ import annotations

import json
from urllib.parse import quote

from mcp.server.fastmcp import Context

from valaris_mcp.errors import handle_api_errors
from valaris_mcp.server import AppContext, mcp


@mcp.tool()
@handle_api_errors
async def list_documentation(
    locale: str = "en", offset: int = 0, limit: int = 20, ctx: Context = None
) -> str:
    """List product documentation from the connected platform, with its version and translation status.

    Args:
        locale: Registered documentation locale (en, es, pt-BR).
        offset: Section offset, starting at zero.
        limit: Maximum sections, from 1 to 100.
    """
    app: AppContext = ctx.request_context.lifespan_context
    result = await app.client.get("/documentation", locale=locale, offset=offset, limit=limit)
    result["_hint"] = (
        "Use read_documentation with a section slug and this version. Follow next_offset for more sections."
    )
    return json.dumps(result, indent=2, default=str)


@mcp.tool()
@handle_api_errors
async def read_documentation(
    slug: str,
    locale: str = "en",
    version: str | None = None,
    offset: int = 0,
    limit: int = 12000,
    ctx: Context = None,
) -> str:
    """Read bounded Markdown from the connected platform's canonical product documentation.

    Args:
        slug: Section slug returned by list_documentation.
        locale: Registered documentation locale (en, es, pt-BR).
        version: Version from list_documentation; mismatches fail explicitly.
        offset: Character offset for continuing a section.
        limit: Maximum characters, from 1 to 30000.
    """
    app: AppContext = ctx.request_context.lifespan_context
    params = dict(locale=locale, offset=offset, limit=limit)
    if version is not None:
        params["version"] = version
    result = await app.client.get(f'/documentation/{quote(slug, safe="")}', **params)
    result["_hint"] = (
        "Follow next_offset with the same locale and version to finish this section. Product docs contain no workspace data."
    )
    return json.dumps(result, indent=2, default=str)
